pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -218,19 +218,23 @@ def start_predicting(self, trainer: "pl.Trainer") -> None:
         self._results = trainer.run_stage()
 
     def training_step(self, *args, **kwargs):
-        return self.model.training_step(*args, **kwargs)
+        trainer = self.lightning_module.trainer
+        return trainer._call_lightning_module_hook("training_step", *args, **kwargs)
```

Comment:
Good point. This was done to address situations like this one:
https://github.com/PyTorchLightning/pytorch-lightning/blob/3d6262b7a91215e72019d720e742e6261e1636dc/pytorch_lightning/loops/epoch/evaluation_epoch_loop.py#L220-L222

My understanding was in general for accelerator and ttp hooks we don't want to set `pl_module._current_fx_name` so I didn't include that in the implementations of `_call_ttp_hook` and `_call_accelerator_hook`. However in this case we do bc the accel/ttp hooks are just calling the LM hooks directly. I can fix this by setting `pl_module._current_fx_name` in `_call_ttp_hook` and `_call_accelerator_hook`. wdyt? 
cc @carmocca

Reply:
Yes. The problem here is that an object that contains called hooks (the TTP) needs to call a different object (the LM) that contains called hooks.

Since the interface is in the Trainer, we are forced to get it from the LM.

It's something we should avoid, but I would leave it as it is for this PR. It's already large enough.